hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/MembershipManager.java

```diff
@@ -456,27 +487,37 @@ void setMembers(MemberMap memberMap) {
 
     // called under cluster service lock
     private void retainSuspectedMembers(MemberMap memberMap) {
-        Iterator<Address> it = suspectedMembers.iterator();
+        Iterator<MemberImpl> it = suspectedMembers.iterator();
         while (it.hasNext()) {
-            Address suspectedAddress = it.next();
-            if (!memberMap.contains(suspectedAddress)) {
+            Member suspectedMember = it.next();
+            if (memberMap.getMember(suspectedMember.getAddress(), suspectedMember.getUuid()) == null) {
                 if (logger.isFineEnabled()) {
-                    logger.fine("Removing suspected address " + suspectedAddress + ", it's no longer a member.");
+                    logger.fine("Removing suspected address " + suspectedMember.getAddress() + ", it's no longer a member.");
                 }
 
                 it.remove();
             }
         }
     }
 
-    boolean isMemberSuspected(Address address) {
-        return suspectedMembers.contains(address);
+    Collection<MemberImpl> getSuspectedMembers() {
+        return new HashSet<>(suspectedMembers);
+    }
+
+    boolean isMemberSuspected(Address address, UUID uuid) {
+        for (MemberImpl suspectedMember : suspectedMembers) {
+            if (suspectedMember.getAddress().equals(address) && suspectedMember.getUuid().equals(uuid)) {
+                return true;
+            }
+        }
+
+        return false;
     }
 
-    boolean clearMemberSuspicion(Address address, String reason) {
+    boolean clearMemberSuspicion(Address address, UUID uuid, String reason) {
```

Comment:
Can't we make just use `Member` as parameter, instead of `Address` and `UUID` pair?

`clearMemberSuspicion(MemberImpl member, String reason)`